test/unclogged/core_test.clj

```diff
@@ -209,3 +210,44 @@
       :rfc-5424 MessageFormat/RFC_5424
       :rfc_5424 MessageFormat/RFC_5424
       :rfc5424 MessageFormat/RFC_5424)))
+
+(deftest ->syslog-msg-tests
+  (testing "no defaults, all keys"
+    (let [contents {:message "hello"
+                    :message-id "xyzzy"
+                    :app-name "unclogged"
+                    :hostname "ditka"
+                    :process-id "1234"}
+          syslog-msg (@#'unclogged.core/->syslog-msg {} contents)]
+      (is (= "hello" (.toString ^CharArrayWriter
+                                (.getMsg ^SyslogMessage syslog-msg))))
+      (is (= "xyzzy" (.getMsgId ^SyslogMessage syslog-msg)))
+      (is (= "unclogged" (.getAppName ^SyslogMessage syslog-msg)))
+      (is (= "ditka" (.getHostname ^SyslogMessage syslog-msg)))
+      (is (= "1234" (.getProcId ^SyslogMessage syslog-msg)))))
+  (testing "some defaults, all keys"
+    (let [defaults {:app-name "unclogged"
+                    :hostname "ditka"
+                    :process-id "1234"}
+          contents {:message "hello"
+                    :message-id "xyzzy"}
+          syslog-msg (@#'unclogged.core/->syslog-msg defaults contents)]
```

Comment:
That's slightly different. `#'` is varquote, so `#'unclogged.core/facility` is not the same thing as the actual private function. One is a var (reference); the dereffed expr is the function itself. I'm not sure which is preferred here; I'll ask around. This sometimes matters; e.g. when you're rebinding, you might want the non-dereffed version since the reference will be resolved before you eval that expr so you'll continue hitting the original (pre-rebinding) version.